tests/compiler/resolve-binary.optimized.wat

```diff
@@ -1564,7 +1564,9 @@
    call $~lib/builtins/abort
    unreachable
   end
-  i32.const 1
+  i32.const 160
+  i32.const 160
+  call $~lib/string/String.__eq
```

Comment:
Hmm. Seems this force full string equal

Reply:
Yeah, the downside is that we cannot precompute `===` string comparisons anymore. Previously, `===` compiled to just an `i32.eq` due to the differing semantics, which is trivial to precompute, while it now calls the overload.

Reply:
Got it. Hope increase limits for inlining helps with this

Reply:
Makes me wonder if we can handle strings differently, by doing something like `i32.eq(lhs, rhs) | String.__eq(lhs, rhs)` where Binaryen can see that this is trueish and potentially eliminate the overload call.

Reply:
Hmm, I prefer use overloading for string operators. It simplify compiler and the same time show more explicitly what's doing on. Also it allow inherit String class and overload any operator which also pretty flexible.

I guess tuning inline params need anyway and this will helps with this. Also it could be done earlier in MIR part in future